test/unit/metrics/test_metricframe_streaming.py

```diff
@@ -0,0 +1,215 @@
+    batch_generator = batchify(num_batches, y_t, y_p, g_4)
+    target = metrics.MetricFrame(skm.recall_score, [], [], sensitive_features=[],
+                                 streaming=True)
+
+    for y_t_batches, y_p_batches, g_4_batches in batch_generator:
+        g_f = pd.DataFrame(data=g_4_batches, columns=['My feature'])
+        target.add_data(transform_y_t(y_t_batches),
+                        transform_y_p(y_p_batches),
+                        sensitive_features=g_f)
+
+    # Check on the indices properties
+    assert target.control_levels is None
+    assert isinstance(target.sensitive_levels, list)
+    assert (target.sensitive_levels == ['My feature'])
+
+    # Check we have correct return types
+    assert isinstance(target.overall, float)
+    assert isinstance(target.by_group, pd.Series)
+
+    # Check we have expected number of elements
+    assert len(target.by_group) == 2
+    assert np.array_equal(target.by_group.index.names, ['My feature'])
+
+    # Compare to the non-batched version.
+    g4 = pd.DataFrame(data=g_4, columns=['My feature'])
+    met_regular = metrics.MetricFrame(skm.recall_score, y_t, y_p, sensitive_features=g4)
+    assert target.overall == met_regular.overall
+
+    for k, v in met_regular.by_group.items():
+        assert v == target.by_group[k]
+
+    assert target.group_min() == met_regular.group_min()
+    assert target.group_max() == met_regular.group_max()
+
+    # If we continue to add data, it resets the metrics
+    g_f = pd.DataFrame(data=g_4_batches, columns=['My feature'])
+    target.add_data(transform_y_t(y_t_batches),
+                    transform_y_p(y_p_batches),
+                    sensitive_features=g_f)
+    assert target._overall is None
+    assert target._by_group is None
+
+    # We can't initialize streaming metrics with data
+    with pytest.raises(ValueError):
+        _ = metrics.MetricFrame(skm.recall_score, y_t, y_p,
+                                sensitive_features=g4, control_features=g4,
+                                streaming=True)
+
+
+def test_exception_on_no_data():
+    target = metrics.MetricFrame(skm.recall_score, [], [], sensitive_features=[],
+                                 sample_params=None, streaming=True)
+
+    with pytest.raises(ValueError) as excinfo:
+        _ = target.control_levels
+    assert "No data to process" in str(excinfo.value)
+
+
+@pytest.mark.parametrize("transform_y_p", conversions_for_1d)
+@pytest.mark.parametrize("transform_y_t", conversions_for_1d)
+@pytest.mark.parametrize("num_batches", [1, 3, 5])
+def test_cf_argument_mismatch(transform_y_t, transform_y_p, num_batches):
+    batch_generator = batchify(num_batches, y_t, y_p, g_4)
+
+    # We initialize without CF
+    target = metrics.MetricFrame(skm.recall_score, [], [], sensitive_features=[],
+                                 streaming=True)
+    # We supply CF in the call, it should fail.
+    with pytest.raises(ValueError):
```

Comment:
Should there be a check on the exception message, so you can be sure that it raised on the right statement?